tools/build_langserver/langserver/completion.go

```diff
@@ -26,26 +27,19 @@ func (h *LsHandler) handleCompletion(ctx context.Context, req *jsonrpc2.Request)
 	if err := json.Unmarshal(*req.Params, &params); err != nil {
 		return nil, err
 	}
-	log.Info("completion with param %s", req.Params)
-	documentURI, err := EnsureURL(params.TextDocument.URI, "file")
+
+	documentURI, err := getURIAndHandleErrors(params.TextDocument.URI, completionMethod)
 	if err != nil {
-		return nil, &jsonrpc2.Error{
-			Code:    jsonrpc2.CodeInvalidParams,
-			Message: fmt.Sprintf("invalid documentURI '%s' for method %s", documentURI, completionMethod),
-		}
-	}
-	if !h.analyzer.IsBuildFile(documentURI) {
-		return nil, &jsonrpc2.Error{
-			Code:    jsonrpc2.CodeInvalidParams,
-			Message: fmt.Sprintf("documentURI '%s' is not supported because it's not a buildfile", documentURI),
-		}
+		return nil, err
 	}
 
-	supportSnippet := h.init.Capabilities.TextDocument.Completion.CompletionItem.SnippetSupport
-
 	h.mu.Lock()
-	itemList, err := getCompletionItemsList(ctx, h.analyzer, supportSnippet, documentURI, params.Position)
-	h.mu.Unlock()
+	itemList, err := h.getCompletionItemsList(ctx, documentURI, params.Position)
+	defer h.mu.Unlock()
```

Comment:
defer immediately after the call to `Lock`. That way if `getCompletionItemsList` panics it will still unlock.